src/main/java/seedu/address/model/country/Country.java

```diff
@@ -1,16 +1,69 @@
 package seedu.address.model.country;
 
-public enum Country {
-//    TODO: Add implementation
-//    private Set<Note> noteList;
-//    public Set<Note> getNoteList() {
-//        if (noteList != null) {
-//            return Collections.unmodifiableSet(noteList);
-//        } else {
-//            return new HashSet<>();
-//        }
-//    }
-//    public void setNoteList(Set<Note> noteList) {
-//        this.noteList = noteList;
-//    }
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+
+import seedu.address.model.note.Note;
+
+/**
+ * A representation of a Country that can be identified by a 2-letter ISO3166 country-code or by its country
+ * name. It contains a list of country notes.
+ */
+public class Country {
+
+    private final String countryName;
+    private final String countryCode;
+    private final ArrayList<Note> countryNotes;
+
+    /**
+     * Initializes a Country by its countryCode.
+     *
+     * @param countryCode The ISO3166 2-letter country code of the country to be initialized.
+     */
+    public Country(String countryCode) {
+        this.countryCode = countryCode;
+        this.countryName = new Locale("", countryCode).getDisplayName();
```

Comment:
```suggestion
        this.countryName = new Locale("", countryCode).getDisplayCountry();
```
I think its more intuitive to call this specific method